packages/components/carousel/src/use-carousel-item.ts

```diff
@@ -67,7 +68,12 @@ export const useCarouselItem = (
   }
 
   function calcCardTranslate(index: number, activeIndex: number) {
-    const parentWidth = carouselContext.root.value?.offsetWidth || 0
+    let parentWidth = carouselContext.root.value?.offsetWidth || 0
+
+    if (unref(isVertical)) {
+      parentWidth = carouselContext.root.value?.offsetHeight || 0
+    }
```

Comment:
Can this code be optimized?

Reply:
ok, I changed it to this 
<img width="509" alt="image" src="https://github.com/element-plus/element-plus/assets/26755049/745744e1-7c82-4546-8126-7b3d81912f7e">
@tolking